### PR TITLE
[ParticleMechanics] Partial migration to amatrix

### PR DESCRIPTION
**Pull request by bodhinandach**
@philbucher I checked without including Amatrix and all tests are working in FullDebug.
I think there is something wrong and not complete yet in Amatrix.
So I will recheck it again once all correction is done, and if necessary, another PR.

### CODE REVIEW

File: applications/ParticleMechanicsApplication/custom_constitutive/hyperelastic_3D_law.cpp
Comment (philbucher) [inline applications/ParticleMechanicsApplication/custom_constitutive/hyperelastic_3D_law.cpp:663]: you cannot resize a bounded matrix!

**Review by philbucher — DISMISSED** [review]
to me the changes look ok

the errors you are getting could come from aliasing
maybe the others can have a look as well

File: applications/ParticleMechanicsApplication/custom_constitutive/hyperelastic_3D_law.cpp
Comment (philbucher) [reply applications/ParticleMechanicsApplication/custom_constitutive/hyperelastic_3D_law.cpp:663]: => resize happens in the next line
@pooyan-dadvand shouldn't this throw a compiler error?

Comment (philbucher) [reply applications/ParticleMechanicsApplication/custom_constitutive/hyperelastic_3D_law.cpp:663]: sorry I misread the code, nevermind

**Comment by bodhinandach** [comment]
@jcotela The remaining errors are, I think, related to the SparseMatrix or Vector. Which I will wait once you and @pooyan-dadvand finalized it. Please keep me updated.

**Comment by pooyan-dadvand** [comment]
I would suggest you to complete the migration keeping the sparse matrix and vector of ublas because the AMatrix sparse would not arrive to this release....

**Comment by jcotela** [comment]
@pooyan-dadvand The remaining errors @bodhinandach mentions is the problem I was discussing with you on Friday. The issue seems to be caused by the removal of the python interface for the ublas vector, which was still being used to define a sparse vector for some tests in this application.

**Comment by pooyan-dadvand** [comment]
Sorry, I have not seen it! 

In AMatrix they will start to be the same vector but the final objective would be to add a NumaVector which has parallelization inside (not like ublas). Maybe we should consider the idea of @philbucher in getting the vector from UblasSpace.

**Comment by jcotela** [comment]
@pooyan-dadvand we tried doing that, but the problem is that the return type of the ublas space is an ublas vector, and pybind complains because there is no interface to hold it.

**Comment by pooyan-dadvand** [comment]
They are only tests that affected by this change? We may disable them temporary until I finish with CSR Matrix...

**Comment by jcotela** [comment]
I think so. @bodhinandach can you confirm?

**Comment by bodhinandach** [comment]
@pooyan-dadvand @jcotela I think I will not disable them, but instead, I will put a FIXME flag so that I remember to get to them again after the complete migration is done, I mean when you remove the Ublas completely. Meanwhile, I might need them to be there as some people will start working to the development of ParticleMechanics and so that the tests are not completely abandoned.

**Comment by bodhinandach** [comment]
@philbucher @jcotela would you mind rechecking this again. If this PR is okay, I am waiting for an approval. Thanks :+1:

**Review by jcotela — APPROVED** [review]

**Pull request merged**
